### ISSUE TITLE
Illegal Instruction error thrown when calling from command-line on OS X

### ISSUE THREAD

**Issue by jthomas** [cloudant/swift-cloudant#170]
I've found a bug which only surfaces when using the library from the command-line using `swift run` on OS X, rather than XCode.

Here's my `main.swift`...

```swift
import SwiftCloudant
import Foundation

let url = URL(string: "http://localhost:5984")!
let client = CouchDBClient(url:url, username:"username", password:"password")
let dbName = "todos"

let allDocs = GetAllDocsOperation(databaseName: "todos", rowHandler: { doc in
    print("Got document: \(doc)")
}) { response, info, error in
    if let error = error {
        print("error", error)
    } else {
        print("success")
    }
}

client.add(operation: allDocs)
CFRunLoopRun()
```

Here's the package manifest
```swift
// swift-tools-version:4.0
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

let package = Package(
    name: "cloudant",
    dependencies: [
        // Dependencies declare other packages that this package depends on.
        .package(url: "https://github.com/cloudant/swift-cloudant.git", .revision("8bf984ef9f4040dcb50169c57769888f71f4fc38")),        
    ],
    targets: [
        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
        // Targets can depend on other targets in this package, and on products in packages which this package depends on.
        .target(
            name: "cloudant",
            dependencies: ["SwiftCloudant"]),
    ]
)
```

If I run the code using `swift run` the following error is thrown.

```
$ swift run
Compile Swift Module 'SwiftCloudant' (25 sources)
Compile Swift Module 'cloudant' (1 sources)
Linking ./.build/x86_64-apple-macosx10.10/debug/cloudant
Fatal error: Unexpectedly found nil while unwrapping an Optional value
Illegal instruction: 4
```

If I set up an xcode project for this test code and start the binary through that it works, which was super confusing! 

Digging into the library, I found the cause of the issue.
https://github.com/cloudant/swift-cloudant/blob/master/Source/SwiftCloudant/HTTP/URLSession.swift#L467-L468

```swift
bundleDisplayName = frameworkBundle.object(forInfoDictionaryKey: "CFBundleName") as! String
bundleVersionString = frameworkBundle.object(forInfoDictionaryKey: "CFBundleShortVersionString") as! String
```

`frameworkBundle.object` returns nil when running from the command-line.

I've opened a PR to resolve this issue.

**Comment by tomblench** [comment]
@jthomas thanks for digging into this but I don't think that it makes sense for us to load the version number and name from a plist. It also seems like the version from the plist has been incorrect and therefore has been reporting the version as 1.0 instead of 0.7.0 on macOS for some time.

I have raised https://github.com/cloudant/swift-cloudant/pull/173 as an alternative approach which will hopefully be merged soon.

**Comment by jthomas** [comment]
Thanks for looking into this. You can close #171?

**Comment by tomblench** [comment]
@jthomas now that #173 has merged, can you check that this fix works, or are you waiting on an "official" release?

**Comment by ricellis** [comment]
This should be resolved in the [0.8.0 release](https://github.com/cloudant/swift-cloudant/releases/tag/0.8.0).

**Comment by jthomas** [comment]
@tomblench Thanks for resolving this, I've verified it now works.
